Crash when sending stats to proxy

### Expected Behavior

Doesn't crash
### Current Behavior

```
ERROR[12-07|19:45:05.030] Failed to decode message from payload    address=0x6729622cB0535f168e744c4ecBc380fb230cB9b2 func=handleMsg cur_seq=1 cur_round=0 state="Waiting for new round" address=0x6729622cB0535f168e744c4ecBc380fb230cB9b2 err="unauthorized address"
INFO [12-07|19:45:05.919] Block synchronisation started 
INFO [12-07|19:45:05.919] Mining aborted due to sync 
WARN [12-07|19:45:05.920] Synchronisation failed, retrying         err="block download canceled (requested)"
ERROR[12-07|19:45:05.922] Failed to set randomness for proposer selection block_number=0 hash=81cb03…f694af error="contract registry not deployed"
WARN [12-07|19:45:05.924] Cannot read block gas limit              err="contract registry not deployed"
INFO [12-07|19:45:05.926] Commit new mining work                   number=1 sealhash=dbffbc…4b36f1 uncles=0 txs=0 gas=0 fees=0 elapsed=3.332ms
WARN [12-07|19:45:05.927] Block sealing failed                     err=unauthorized
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xb7666e]

goroutine 105 [running]:
github.com/ethereum/go-ethereum/consensus/istanbul/backend.(*Backend).SendDelegateSignMsgToProxy(0xc0000ee000, 0xc02a56c300, 0x2f7, 0x300, 0x300, 0x0)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/consensus/istanbul/backend/backend.go:210 +0x1ae
github.com/ethereum/go-ethereum/ethstats.(*Service).handleDelegateSign(0xc0000f6c00, 0xc02a59a1c0, 0x0, 0x2)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:474 +0x121
github.com/ethereum/go-ethereum/ethstats.(*Service).loop(0xc0000f6c00)
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:290 +0x3c4
created by github.com/ethereum/go-ethereum/ethstats.(*Service).Start
        /go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/ethstats/ethstats.go:160 +0x56
```

--- a/consensus/istanbul/backend/backend.go
+++ b/consensus/istanbul/backend/backend.go
@@ -196,7 +196,7 @@ func (sb *Backend) IsProxy() bool {
 }
 
 func (sb *Backend) IsProxiedValidator() bool {
-	return sb.proxyNode != nil
+	return sb.proxyNode != nil && sb.proxyNode.peer != nil
 }
 
 // SendDelegateSignMsgToProxy sends an istanbulDelegateSign message to a proxy

--- a/ethstats/ethstats.go
+++ b/ethstats/ethstats.go
@@ -445,7 +445,7 @@ func (s *Service) login(conn *websocket.Conn, sendCh chan *StatsPayload) error {
 	// Retrieve the remote ack or connection termination
 	var ack map[string][]string
 	if err := websocket.JSON.Receive(conn, &ack); err != nil {
-		return err
+		return errors.New("unauthorized, try registering your validator to get whitelisted")
 	}
 	emit, ok := ack["emit"]
 	if !ok {

